Index: octocatalog-diff/lib/octocatalog-diff/catalog-diff/differ.rb
===================================================================
--- octocatalog-diff.orig/lib/octocatalog-diff/catalog-diff/differ.rb
+++ octocatalog-diff/lib/octocatalog-diff/catalog-diff/differ.rb
@@ -11,6 +11,8 @@ require_relative '../errors'
 require_relative '../util/util'
 require_relative 'filter'
 
+HashDiff = Hashdiff unless defined? HashDiff
+
 module OctocatalogDiff
   module CatalogDiff
     # Calculate the difference between two Puppet catalogs.
